lib/nbtree/nbtsort.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * pg_bulkload: lib/nbtree/nbtsort.c
+ *
+ * NOTES
+ *
+ * Although nbtsort-14.c is the copy of postgresql core's src/backend/access/nbtree/nbtsort.c,
+ * this file has functions which related to nbtsort, but is not implemented core's code.
+ *
+ *	  Copyright (c) 2021, NIPPON TELEGRAPH AND TELEPHONE CORPORATION
+ */
+
+
+#if PG_VERSION_NUM >= 140000
+/*
+ * create and initialize a spool structure
+ */
+static BTSpool *
+_bt_spoolinit(Relation heap, Relation index, bool isunique, bool isdead)
```

Comment:
Is there a reason why your opinion moved the "_bt_spoolinit" function outside?
In my think, it's more consistent with adding to "nbtsort-14.c" like what we've worked on it so far.
If you think it is good to add the "_bt_spoolinit" function to "nbtsort.c
", I think it would be good to move all "_bt_spoolinit" functions included to "nbtsort-XX.c" to "nbtsort.c".

What do you think of this opinion?

Reply:
I thought it's better for a maintenance reason.

I puzzled that the header comments of "nbtsort-XX.c" say the identification is "src/backend/access/nbtree/nbtsort.c", but there is a fucntion which the PostgreSQL don't have in the version. 

I though it's consistent  if "nbtsort-XX.c" is just copy of postgresql's "src/backend/access/nbtree/nbtsort.c".


> If you think it is good to add the "_bt_spoolinit" function to "nbtsort.c
", I think it would be good to move all "_bt_spoolinit" functions included to "nbtsort-XX.c" to "nbtsort.c".

OK, but why don't you after merging this commit? Because _bt_spoolinit was PostgreSQL's code, for example, 9.6 has that. After merging this, I'll update all nbtsort-XX.c and if there are other functions for only pg_bulkload, I'll move to "nbtsort.c" together.

Reply:
OK. For that reason, I agree with your opinion.
> OK, but why don't you after merging this commit? 

Also, I understand moving the "_bt_spoolinit" funciton and 
anyone from "nbtsort-xx.c" to "nbtsort.c" after committing the current PR.

> BTW, should "nbtsort.c" in pg_bulkload be renamed though I didn't come up with any good name? This makes maintainers confused, doesn't it?

I think that "nbtsort-common.c" might be good 
because this file includes a common functions.
Of course, this is my personal opinion, so I think it's fine as it is. :)

Best regards.
Moon.

Reply:
Thanks, OK

> Also, I understand moving the "_bt_spoolinit" funciton and
anyone from "nbtsort-xx.c" to "nbtsort.c" after committing the current PR.

I made #106.

> I think that "nbtsort-common.c" might be good
because this file includes a common functions.
Of course, this is my personal opinion, so I think it's fine as it is. :)

It looks good to me. I changed the name.

Reply:
Thanks for fixing the file name and creating a new issue!
I think there's nothing problme with committing this PR.
That is, it looks good to me.